packages/delegate/src/prepareGatewayDocument.ts

```diff
@@ -442,8 +444,13 @@ function wrapConcreteTypes(
         }
       },
       [Kind.FIELD]: (node: FieldNode) => {
-        const type = typeInfo.getType();
-        if (type != null && isAbstractType(getNamedType(type))) {
+        let type = typeInfo.getType();
+        type = isNonNullType(type) ? type.ofType : type;
+        if (
+          type != null &&
+          isAbstractType(getNamedType(type)) &&
```

Comment:
Why do we need to spread on abstract type ?
Shouldn't the spread be inside the selection set of this type, to spread inline fragment for each final types possible ?